### PR TITLE
update cncf conformance job

### PR DESCRIPTION
**Pull request by kwmonroe**
- new job name to group all our conformance jobs together
- run weekly
- sonobuoy needs allow-privileged; make sure that's set in the overlay
- capture the sonobuoy tarball in our results
- push what we know to our cncf fork to ease upstream PR creation
- use latest sonobuoy version (0.56.3 atm)

Note: We'll need to configure the `conformance-cncf-ck` job and switch the scm branch to `main` after merge.

### CODE REVIEW

**Comment by kwmonroe** [comment]
Good run with this branch:

https://jenkins.canonical.com/k8s/job/conformance-cncf-ck/6/

PR-prep logic worked to make the upstream submission pretty easy:

https://github.com/cncf/k8s-conformance/pull/1846

File: jobs/cncf-conformance/conformance-spec [hunk 1]
```diff
@@ -16,54 +16,110 @@ set -x
...
+    rm -f ${SB_FILE}
+    ./sonobuoy version || exit 1
+}
+
+function juju::deploy::overlay
+{
+    cat <<EOF > overlay.yaml
+series: $SERIES
+applications:
+  kubernetes-master:
```
Comment (addyess) [inline jobs/cncf-conformance/conformance-spec:32]: inclusive naming?

```suggestion
  kubernetes-control-plane:
```

File: jobs/cncf-conformance/conformance-spec [hunk 2]
```diff
@@ -16,54 +16,110 @@ set -x
...
 }
 
 function test::execute
 {
     declare -n is_pass=$1
 
     mkdir -p $HOME/.kube
-    juju scp -m $JUJU_CONTROLLER:$JUJU_MODEL kubernetes-master/0:config $HOME/.kube/
-    export RBAC_ENABLED=$(kubectl api-versions | grep \"rbac.authorization.k8s.io/v1beta1\" -c)
+    juju scp -m $JUJU_CONTROLLER:$JUJU_MODEL kubernetes-master/leader:config $HOME/.kube/
```
Comment (addyess) [inline jobs/cncf-conformance/conformance-spec:47]:
```suggestion
    juju scp -m $JUJU_CONTROLLER:$JUJU_MODEL kubernetes-control-plane/leader:config $HOME/.kube/
```

File: jobs/cncf-conformance/conformance-spec [hunk 3]
```diff
@@ -16,54 +16,110 @@ set -x
...
-JUJU_DEPLOY_BUNDLE=cs:~containers/charmed-kubernetes
-JUJU_DEPLOY_CHANNEL=${3:-stable}
 JUJU_CLOUD=aws/us-east-1
-JUJU_CONTROLLER=validate-$(identifier::short)
-JUJU_MODEL=validate-ck
-ARCH=${4:-amd64}
-JOB_NAME_CUSTOM="validate-ck-conformance-$SERIES-$SNAP_VERSION"
+JUJU_CONTROLLER=cncf-ck-$(identifier::short)
+JUJU_DEPLOY_BUNDLE=charmed-kubernetes
+JUJU_DEPLOY_CHANNEL=stable
```
Comment (addyess) [inline jobs/cncf-conformance/conformance-spec:112]: i'm guessing we can't be inclusively named yet b/c the bundle still points at `kubernetes-master`

**Review by addyess — APPROVED** [review]
Looking good.  Maybe parameterize for when the charm master->control-plane shifts

**Comment by kwmonroe** [comment]
I merged this since we have good runs. I don't want to bother param'ing the charm name because once we switch to 1.24, the charm name will stabilize.  We don't run conformance again on past releases.

I've opened this to track the name change once we go for 1.24 conformance:

https://bugs.launchpad.net/charmed-kubernetes-testing/+bug/1967772

**Pull request merged**